File: go/anagram/anagram.go
package anagram

import "strings"

func Detect(subject string, candidates []string) []string {
	var result []string
	lowerCaseSubject := strings.ToLower(subject)

	for _, word := range candidates {
		if isAnagram([]rune(lowerCaseSubject), []rune(strings.ToLower(word))) {
			result = append(result, word)
		}
	}
	return result
}

func isAnagram(target, candidate []rune) bool {
	if string(target) == string(candidate) {
		return false
	}
	targetCount := runeCount(target)
	candidateCount := runeCount(candidate)

	if len(targetCount) != len(candidateCount) {
		return false
	}
	for char, count := range targetCount {
		if count != candidateCount[char] {
			return false
		}
	}
	return true
}

func runeCount(runeSlice []rune) map[rune]int {
	rc := map[rune]int{}

	for _, r := range runeSlice {
		rc[r]++
	}
	return rc
}

File: python/anagram/anagram.py
def find_anagrams(word: str, candidates: list) -> set:
    anagrams = []
    for candidate in candidates:
        if char_frequency(word) == char_frequency(candidate) and candidate.lower() != word.lower():
            anagrams.append(candidate)
    return anagrams

def char_frequency(word: str) -> dict:
    frequency = {}
    lower_word = word.lower()
    for char in lower_word:
        if char not in frequency:
            frequency[char] = 0
        frequency[char] += 1
    return frequency